source/simulator/assemblers/advection.cc

```diff
@@ -613,6 +613,16 @@ namespace aspect
               this->get_material_model().evaluate(scratch.neighbor_face_material_model_inputs,
                                                   scratch.neighbor_face_material_model_outputs);
 
+              if (parameters.formulation_temperature_equation ==
+                  Parameters<dim>::Formulation::TemperatureEquation::reference_density_profile)
+                {
+                  const unsigned int n_q_points = scratch.neighbor_face_finite_element_values->n_quadrature_points;
```

Comment:
I think the ``n_q_points`` variable in the outer scope would be correct too, so delete this line.